### PR TITLE
apt_rpm: add check if package version is upgradable (ansible-collections#7414)

### PR DESCRIPTION
**Pull request by burykinne**
##### SUMMARY
Added function to compare version between installed package and candidate. 
Package will be installed if the candidate is newer than the existing.

Fixes https://github.com/ansible-collections/community.general/issues/7414.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

plugins/modules/apt_rpm.py

### CODE REVIEW

**Comment by felixfontein** [comment]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

**Comment by burykinne** [comment]
> Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Added a changelog fragment. I should have done it earlier, sorry.

File: changelogs/fragments/7577-fix-apt_rpm-module.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - apt-rpm - not upgrade packages if a newer version exists. Now package will be installed if the candidate is newer than the existing (https://github.com/ansible-collections/community.general/issues/7414).
```
Comment (felixfontein) [inline changelogs/fragments/7577-fix-apt_rpm-module.yml:2]:
```suggestion
  - apt-rpm - the module did not upgrade packages if a newer version exists. Now the package will be reinstalled if the candidate is newer than the installed version (https://github.com/ansible-collections/community.general/issues/7414).
```

File: plugins/modules/apt_rpm.py [hunk 2]
```diff
@@ -165,6 +167,21 @@ def query_package(module, name):
         return False
 
 
+def check_package_version(module, name):
+    # compare installed and candidate version
+    # if newest version already installed return True
+    # otherwise return False
+
+    rc, out, err = module.run_command("%s policy %s" % (APT_CACHE, name), \
```
Comment (felixfontein) [inline plugins/modules/apt_rpm.py:175]: It is better to use the list version of `run_command()` than the string version:
```suggestion
    rc, out, err = module.run_command([APT_CACHE, "policy", name], \
```

File: plugins/modules/apt_rpm.py [hunk 3]
```diff
@@ -179,8 +196,11 @@ def query_package_provides(module, name):
         name = local_rpm_package_name(name)
 
     rc, out, err = module.run_command("%s -q --provides %s" % (RPM_PATH, name))
-    return rc == 0
-
+    if rc == 0:
+        if check_package_version(module, name):
+            return True
+    else:
+        return False
```
Comment (felixfontein) [inline plugins/modules/apt_rpm.py:203]: Right now if `rc == 0` but your function returns a falsish value, this function returns `None`. Better change this to
```suggestion
    return False
```
to return `False` instead.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by felixfontein** [comment]
(Please note that you also have to fix the sanity checks.)

**Comment by burykinne** [comment]
> (Please note that you also have to fix the sanity checks.)

Thanks for your comments. I'm going to fix it.

**Comment by felixfontein** [comment]
Looks good to me, as far as I can judge this. If nobody objects, I'll merge this in ~one week.

**Comment by felixfontein** [comment]
@burykinne thanks a lot for your contribution!

**Pull request merged**